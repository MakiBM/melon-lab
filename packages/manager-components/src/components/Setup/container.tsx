import { withFormik } from 'formik';
import { compose, withHandlers } from 'recompose';
import * as Yup from 'yup';
import Setup from './index';

const withFormValidation = withFormik({
  mapPropsToValues: props => ({ ...props.initialValues }),
  validationSchema: Yup.object().shape({
    name: Yup.string().required('name is required.'),
  }),
  enableReinitialize: true,
  handleSubmit: (values, form) => {
    if (form.props.onSubmit) {
      form.props.onSubmit(values);
    }
  },
});

const withFormHandler = compose(
  withHandlers({
    onChange: props => (values, event) => {
      props.setFieldValue(event.target.name, values.value);
    },
  }),
);

export default compose(
  withFormValidation,
  withFormHandler,
)(Setup);
