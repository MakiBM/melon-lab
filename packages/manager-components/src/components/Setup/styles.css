.setup {
  max-width: 400px;
  margin: 0 auto var(--space-xl) auto;
}

.setup__info {
  margin-bottom: var(--space-l);
}

.setup__exchanges {
  margin-bottom: var(--space-l);
}

.setup__exchanges-checkbox {
  margin: var(--space-xxs) 0;
}
