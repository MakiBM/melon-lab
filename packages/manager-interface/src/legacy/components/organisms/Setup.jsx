import React from 'react';
import { Field } from 'redux-form';
import { List, Button, Card, Checkbox } from 'semantic-ui-react';

import renderInput from '../utils/renderInput';

// Explicitely decompose props here.
const Setup = ({ loading, handleSubmit, networkId, config }) => (
  <form onSubmit={handleSubmit}>
    <Card centered>
      <Card.Content>
        <div className={`ui ${loading ? 'active' : ''} inverted dimmer`}>
          <div className="ui text loader">
            Deploying your fund to the Ethereum blockchain
          </div>
        </div>

        <br />
        <Card.Header>Setup your fund</Card.Header>

        <List>
          <List.Item>
            <List.Content>
              <Field
                name="name"
                component={renderInput}
                className="left-input"
              />
            </List.Content>
          </List.Item>
          <br />
          <h4>Melon Default Configuration:</h4>
          <p>
            For this version, the modules that your fund will use are predefined
            ie. you do not need to choose a module. For your record, below are
            the predefined modules for this version.
          </p>
          <List.Item>
            <List.Content>Performance fee: 0%</List.Content>
          </List.Item>
          <List.Item>
            <List.Content>Management fee: 0%</List.Content>
          </List.Item>

          <List.Item>
            <List.Content>
              <strong>Exchange:</strong>
              <br />
              <Checkbox label="OasisDex" />
              <br />
              <Checkbox label="ZeroEx" />
              <br />
              <br />
            </List.Content>
          </List.Item>

          <List.Item
            as="a"
            href={`https://${
              networkId === '42' ? 'kovan.' : ''
            }etherscan.io/address/${config.priceFeedAddress}`}
            target="_blank"
          >
            <List.Content>
              Pricefeed: <strong>Thomson Reuters Pricefeed</strong>
            </List.Content>
          </List.Item>

          <List.Item>
            <List.Content>
              Asset Registrar: <strong>Melon Main Net Asset Universe</strong> (4
              assets registered)
            </List.Content>
          </List.Item>
          <List.Item>
            <List.Content
              href={`https://${
                networkId === '42' ? 'kovan.' : ''
              }etherscan.io/address/${config.complianceAddress}`}
              target="_blank"
            >
              Compliance (invest/redeem):{' '}
              <strong>Only manager can invest (in MLN)</strong>
            </List.Content>
          </List.Item>
          <List.Item>
            <List.Content
              href={`https://${
                networkId === '42' ? 'kovan.' : ''
              }etherscan.io/address/${config.riskManagementAddress}`}
              target="_blank"
            >
              Risk Management:{' '}
              <strong>Make/Take order permitted with 10% deviation </strong>{' '}
              from the reference price provided by above pricefeed.
            </List.Content>
          </List.Item>
        </List>

        <Button basic color="black" style={{ width: '100%' }}>
          Create and deploy my fund!
        </Button>
      </Card.Content>
    </Card>
  </form>
);

export default Setup;
