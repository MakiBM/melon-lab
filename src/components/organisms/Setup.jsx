import React from "react";
import { Field } from "redux-form";
import { List, Input, Button, Card } from "semantic-ui-react";

const renderInput = ({ input }) => (
  <Input {...input} placeholder="Fund Name" style={{ width: "100%" }} />
);

// Explicitely decompose props here.
const Setup = ({ loading, name, handleSubmit }) => (
  <form onSubmit={handleSubmit}>
    <Card centered>
      <Card.Content>
        <div className={`ui ${loading ? "active" : ""} inverted dimmer`}>
          <div className="ui text loader">
            Deploying your fund to the Ethereum blockchain
          </div>
        </div>
        <p>Welcome to the future of investment funds</p>
        <p className="App-intro">
          To get started, head to our{" "}
          <strong>
            <a
              href="http://faucet.melon.network"
              target="_blank"
              rel="noopener noreferrer"
            >
              faucet
            </a>
          </strong>{" "}
          to receive Kovan Ether and Kovan Melon
        </p>
        <p className="App-intro">
          Once you have received k-ETH and k-MLN, go ahead and create your Melon
          fund.
        </p>
        <br />
        <Card.Header>Setup your fund</Card.Header>

        <List>
          <List.Item as="a">
            <List.Content>
              <Field name="name" component={renderInput} />
            </List.Content>
          </List.Item>
          <br />
          <h4>Melon Default Configuration:</h4>

          <List.Item>
            <List.Content>Performance fee: 0%</List.Content>
          </List.Item>
          <List.Item>
            <List.Content>Management fee: 0%</List.Content>
          </List.Item>

          <List.Item
            as="a"
            href="https://kovan.etherscan.io/address/0x7f98f4790722a24de32478714f0350a56689825e"
            target="_blank"
          >
            <List.Content>
              Exchange: <strong>OasisDex</strong>
            </List.Content>
          </List.Item>

          <List.Item
            as="a"
            href="https://kovan.etherscan.io/address/0x9ffe1fce6dc97834c5733362d229dfc997299a79"
            target="_blank"
          >
            <List.Content>
              DataFeed: <strong>Thomson Reuters Datafeed</strong>
            </List.Content>
          </List.Item>

          <List.Item>
            <List.Content>
              Asset Registrar: <strong>Melon Asset Universe</strong> (18 assets
              registered)
            </List.Content>
          </List.Item>
          <List.Item>
            <List.Content
              href="https://github.com/melonproject/protocol/blob/0.3.8-alpha.5/contracts/participation/Participation.sol"
              target="_blank"
            >
              Participation: <strong>Subscription in MLN</strong>
            </List.Content>
          </List.Item>
          <List.Item>
            <List.Content
              href="https://github.com/melonproject/protocol/blob/0.3.8-alpha.5/contracts/riskmgmt/RMMakeOrders.sol"
              target="_blank"
            >
              Risk Management: <strong>Make/Take order permitted</strong>
            </List.Content>
          </List.Item>
        </List>

        <Button basic color="black" style={{ width: "100%" }}>
          Create and deploy my fund!
        </Button>
      </Card.Content>
    </Card>
  </form>
);

export default Setup;
