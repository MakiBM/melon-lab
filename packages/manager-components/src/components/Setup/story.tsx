import { storiesOf } from '@storybook/react';
import React from 'react';
import Setup from './container';

const data = {
  initialValues: {
    name: '',
  },
  config: {
    canonicalPriceFeedAddress: 'foo',
    competitionComplianceAddress: 'bar',
    onlyManagerCompetitionAddress: 'foo',
  },
};

storiesOf('Components|Setup', module).add('Default', () => {
  return <Setup {...data} />;
});
