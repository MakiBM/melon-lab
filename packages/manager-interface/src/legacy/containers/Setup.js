import { connect } from 'react-redux';
import { reduxForm } from 'redux-form';
import Setup from '../components/organisms/Setup';
import { actions } from '../actions/fund';

const mapStateToProps = state => ({
  loading: state.app.transactionInProgress,
  networkId: state.ethereum.network,
  config: state.fund.config,
  isCompetition: state.app.isCompetition,
});

const onSubmit = (values, dispatch) => {
  dispatch(actions.setupRequested(values.name, true, true));
};

const SetupRedux = connect(mapStateToProps)(Setup);

const SetupForm = reduxForm({
  form: 'setup',
  onSubmit,
})(SetupRedux);

export default SetupForm;
