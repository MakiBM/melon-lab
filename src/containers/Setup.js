import { connect } from "react-redux";
import Setup from "../components/organisms/Setup";
import { actions } from "../actions/fund";

const mapStateToProps = state => ({
  ...state.setup,
});

const mapDispatchToProps = dispatch => ({
  onCreate: () => {
    dispatch(actions.create());
  },
  onChange: event => {
    dispatch(actions.change({ [event.target.name]: event.target.value }));
  },
});

const SetupContainer = connect(mapStateToProps, mapDispatchToProps)(Setup);

export default SetupContainer;
