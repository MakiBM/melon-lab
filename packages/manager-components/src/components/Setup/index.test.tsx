import React from 'react';
import Setup from './container';

const data = {
  initialValues: {
    name: '',
  },
  config: {
    canonicalPriceFeedAddress: 'foo',
    competitionComplianceAddress: 'bar',
    onlyManagerCompetitionAddress: 'foo',
  },
};

describe('Setup', () => {
  const defaultElement = <Setup {...data} />;

  it('should render correctly', () => {
    const wrapper = shallow(defaultElement);
    expect(wrapper).toMatchSnapshot();
  });
});
