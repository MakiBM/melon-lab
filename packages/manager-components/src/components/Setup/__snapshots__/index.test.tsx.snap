// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Setup should render correctly 1`] = `
<Formik
  config={
    Object {
      "canonicalPriceFeedAddress": "foo",
      "competitionComplianceAddress": "bar",
      "onlyManagerCompetitionAddress": "foo",
    }
  }
  enableReinitialize={true}
  handleSubmit={[Function]}
  initialValues={
    Object {
      "name": "",
    }
  }
  isInitialValid={false}
  onSubmit={[Function]}
  render={[Function]}
  validateOnBlur={true}
  validateOnChange={true}
  validationSchema={[Function]}
/>
`;
